src/posix/platform/system.cpp

```diff
@@ -196,6 +192,61 @@ void platformSetUp(void)
     return;
 }
 
+void otSysBorderRouterInit(otPlatformConfig *aPlatformConfig, int aIcmp6Socket)
+{
+    OT_UNUSED_VARIABLE(aPlatformConfig);
+    OT_UNUSED_VARIABLE(aIcmp6Socket);
+
+#if OPENTHREAD_POSIX_CONFIG_INFRA_IF_ENABLE
+    ot::Posix::InfraNetif::Get().Init(aPlatformConfig->mBackboneInterfaceName, aIcmp6Socket);
+#endif
+
+#if OPENTHREAD_CONFIG_RADIO_LINK_TREL_ENABLE
+    platformTrelInit(getTrelRadioUrl(aPlatformConfig));
+#endif
```

Comment:
TREL is not a Border Router exclusive feature

Reply:
Renamed the APIs to `otSysInfraNetif*`.